packages/widget/src/stores/header/useHeaderStore.tsx

```diff
@@ -35,6 +35,21 @@ export function useHeaderStore<T>(selector: (state: HeaderState) => T): T {
   return useStore(selector);
 }
 
+interface UseHeaderTitleProps {
+  title: string;
+}
+export function useHeaderTitle({ title }: UseHeaderTitleProps) {
+  const headerTitle = useMemo(() => title, [title]);
+  const { setTitle, removeTitle } = useHeaderStore((state) => state);
+
+  useEffect(() => {
+    setTitle(headerTitle);
+    return () => {
+      removeTitle();
+    };
+  }, [setTitle, removeTitle, headerTitle]);
```

Comment:
Maybe `title` can be a string here instead of the object, so we don't need to use `useMemo`?

Reply:
I've made that change and checked the routes again locally 👍